packages/core/src/Platform/index.ts

```diff
@@ -1,30 +1,54 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+import { CustomUserAgentDetails, Framework } from './types';
 import { version } from './version';
+import { detectFramework } from './detectFramework';
+import { UserAgent as AWSUserAgent } from '@aws-sdk/types';
 
-const BASE_USER_AGENT = `aws-amplify/${version}`;
+const BASE_USER_AGENT = `aws-amplify`;
 
+let framework = detectFramework();
+let frameworkHasBeenRerun = false;
 export const Platform = {
-	userAgent: `${BASE_USER_AGENT} js`,
-	product: '',
-	navigator: null,
-	isReactNative: false,
+	userAgent: `${BASE_USER_AGENT}/${version}`,
+	framework,
+	isReactNative: framework === Framework.ReactNative,
 };
-if (typeof navigator !== 'undefined' && navigator.product) {
-	Platform.product = navigator.product || '';
-	Platform.navigator = navigator || null;
-	switch (navigator.product) {
-		case 'ReactNative':
-			Platform.userAgent = `${BASE_USER_AGENT} react-native`;
-			Platform.isReactNative = true;
-			break;
-		default:
-			Platform.userAgent = `${BASE_USER_AGENT} js`;
-			Platform.isReactNative = false;
-			break;
+
+// Rerun framework detection once when getAmplifyUserAgent is called if framework is None:
+// ReactNative framework must be detected initially, however
+// Other frameworks may not be detected in cases where DOM is not yet loaded
+const rerunFrameworkDetection = () => {
+	if (Platform.framework === Framework.None && !frameworkHasBeenRerun) {
+		frameworkHasBeenRerun = true;
+		framework = detectFramework();
+		Platform.framework = framework;
```

Comment:
```suggestion
        framework = detectFramework();
		frameworkHasBeenRerun = true;
		Platform.framework = framework;
```

Nit: Run detection first then update states